services/authorization-service.js

```diff
@@ -0,0 +1,104 @@
+const ERROR = require('../constants/error-constants');
+
+class AuthorizationService {
+
+    constructor(configurationService) {
+        this.configurationService = configurationService;
+    }
+
+    /**
+     * Function to get a user's scopes and scope values for a given permission
+     * @param user {{role: string, permissions: string[]}} - the current user
+     * @param permission {string} - the name of the permission without scopes, please use a permissions constant
+     * for this input
+     * Example: "entity:action"
+     * @returns {{scopes: string[], scopeValues: string[]}} an object containing the user's scopes and scope values for
+     * the input permission. If the user does not have the permission or has no scopes for the permission, then the
+     * default response is {scopes: ["none"], scopeValues: []}
+     */
+    async getPermissionScope(user, permission){
+        let output = {
+            scopes: ['none'],
+            scopeValues: []
+        }
+        const userPermissions = user?.permissions;
+        if (!userPermissions || !permission){
+            return output;
+        }
+        // Loop through the user's permissions until one matching the input permission is found
+        for (const userPermission of userPermissions){
+            const permissionAndScope = parsePermissionString(userPermission);
+            let scopes = permissionAndScope?.scopes;
+
+            if (permissionAndScope?.permission === permission) {
+                if (!!scopes && scopes.length > 0){
+                    output.scopes = scopes;
+                    output.scopeValues = permissionAndScope?.scopeValues || []
```

Comment:
When scope is "study", the scopeValues should be read from user's assigned studies. Similarly, when scope is "DC", scopeValues should be read from user's assigned data commons.